packages/vm/src/modules/in_memory_cache.rs

```diff
@@ -1,9 +1,11 @@
 use clru::CLruCache;
 use wasmer::Module;
 
-use crate::{Checksum, Size, VmResult};
+use crate::{Checksum, Size, VmError, VmResult};
 
-const ESTIMATED_MODULE_SIZE: Size = Size::mebi(10);
+// Rounded-up average of module sizes, in Megabytes.
+// Based on `examples/module_size.sh`, and the cosmwasm-plus modules
+const ESTIMATED_MODULE_SIZE: Size = Size::mebi(5);
```

Comment:
Very unfortunate that we still need the element count

Reply:
The problem is that the cache impl we're using has pre-allocated entries (to achieve O(1) access times), and, so we need to pass (an estimate of) the number of entries.

Only possible improvements I see here:
- Set this number a little lower than the average (`Size::mebi(4)`?). The idea here being, trying to hit the weight limit before the capacity limit. This, at the expense of some wasted memory in the unused entries, i.e. oversized capacity.
- Use another cache impl. One with dynamic capacity. I was taking a look Friday of the different "popular" (more than 1000 downloads) cache crates that are available in crates.io. I'll write a summary / index in another issue, and we can probably revisit this in the future.

Reply:
Another possible strategy:

- Set the number of entries directly, to a large enough value as to render it irrelevant, so that the weight / size cap will always be hit first. And, assume the extra memory cost - deduct it from the total cache mem budget. Given that the memory costs per entry are a few bytes (see https://github.com/maurolacy/clru-rs/blob/d32f5b77823bd99772da32fcb755ff7059377ed1/src/lib.rs#L1062) this is actually **not a bad idea**. If we set capacity to 10000, we could handle a size-limited cache of 2 GB, even when all the entries are ~250 kB, which are the smallest I observed. And, assuming the cost per entry is 48 bytes, that will have an extra cost of just ~500 kB. Which is 0.025 % of our total cache size!

So, I would say, let's do this.

Reply:
I was thinking of this direction as well, and agree we should do it. In https://github.com/CosmWasm/cosmwasm/pull/748 you see the sized of key/value for our usage.

Reply:
OK, I'll do this.
I'll also add a couple performance tests, for a cache operating at or near full size, with random module sizes, to make sure we don't hit performance issues.